spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubProperties.java

```diff
@@ -29,11 +31,18 @@
  */
 @ConfigurationProperties("spring.cloud.gcp.pubsub")
 public class GcpPubSubProperties implements CredentialsSupplier {
-	/** Number of threads used by every {@link com.google.cloud.pubsub.v1.Subscriber}. */
-	private int subscriberExecutorThreads = 4;
 
-	/** Number of threads used by every {@link com.google.cloud.pubsub.v1.Publisher}. */
-	private int publisherExecutorThreads = 4;
+	private final Subscriber subscriber = new Subscriber();
```

Comment:
If we go down this route, I think you'll need `@NestedConfigurationProperties` here and below for `Publisher` too.

Reply:
Mike and Artem are for the inner class , though.

Reply:
We don't need a `@NestedConfigurationProperties` if options classes are inner.